### DISCUSSION TITLE
Data pulled to Power BI

### DISCUSSION THREAD

**Discussion by AJ-Greiner** [nbrown02/FlowViz-Jira#11] [Q&A] [answered]
Good morning, 

I have your connector working but I have a question.

I would like to create a dashboard of my own that includes the custom fields I created (contact type, category and so on). Is there a way to tweak your API to tell it to grab more tables?

Thank you,

**Comment by nbrown02** [comment, marked as answer]
Thanks for the feedback/question @AJ-Greiner. Yes you can do that 👍 

You will have to go into Power Query and some of the previous steps in the data transformation to find the additional columns. 
There is a common step called _Expanded Data.Fields_ where the fields I have used have been expanded:

![image](https://user-images.githubusercontent.com/29369962/176458064-f7883f75-39fa-4ea0-b868-f4c1a4d5ec50.png)

You need to go to the step before that called _Expanded  - this is where you'll find the fields:
![image](https://user-images.githubusercontent.com/29369962/176457834-ad042283-1bc3-474b-bedc-6fc600c415b4.png)

Be mindful you'll need to do this in each table....plus the custom fields may be hard to find - Jira just called them CustomField_nnnn

**Comment by AJ-Greiner** [comment]
Thank you for the quick reply. When I locate the custom fields and then select them it only displays “Record” as the item in the field.
***@***.***

When I look at the data in transform it shows me a value.
***@***.***

I know I have to be doing something wrong. Thanks for your help.


Arthur (AJ) Greiner  |  Help Desk Supervisor
E: ***@***.******@***.***>  |  O: +1.516.283.8640

**Comment by nbrown02** [comment]
Yes you need to keep expanding - 
![image](https://user-images.githubusercontent.com/29369962/176472684-7cf9b601-ac7e-4dfd-afcb-812f599e8840.png)

And select the fields you want - sometimes you'll need to expand to new rows, sometimes extract values. Typically if you get an error after doing that you should go back and do the opposite...

**Comment by AJ-Greiner** [comment]
Got it thanks. If I run into anything I will let you know.



Arthur (AJ) Greiner  |  Help Desk Supervisor
E: ***@***.******@***.***>  |  O: +1.516.283.8640

**Comment by AJ-Greiner** [comment]
Sorry one more question,

Is there a way to get one table that contains all incidents regardless of if they are in progress or completed?

Thanks,

Arthur (AJ) Greiner  |  Help Desk Supervisor
E: ***@***.******@***.***>  |  O: +1.516.283.8640

**Reply by nbrown02** [reply]
Yes - you would just remove anything from the JQL that pertains to status, that way it will bring everything in :)

**Comment by AJ-Greiner** [comment]
Ok thanks again. Have to do some research on doing that.

Arthur (AJ) Greiner  |  Help Desk Supervisor
E: ***@***.******@***.***>  |  O: +1.516.283.8640

**Reply by nbrown02** [reply]
So for example if you did it for the WorkItems Completed table, you would remove the following:
![image](https://user-images.githubusercontent.com/29369962/176523295-51a4affa-8d66-4f93-bb52-ca94c103f67d.png)

You also need to make your queries match with the startat2,3,4 etc. as this is how it does the pagination of results - if they don't match it will bring an error.

If you get stuck, feel free to email me (nicolas.brown@outlook.com) and happy to have a call to help you out :)

**Comment by AJ-Greiner** [comment]
Thanks I appreciate it. I eyes feel like they are going to bleed to I am going to tackle this tomorrow.

Arthur (AJ) Greiner  |  Help Desk Supervisor
E: ***@***.******@***.***>  |  O: +1.516.283.8640
